src/webui/src/components/trial-detail/TableList.tsx

```diff
@@ -45,749 +39,460 @@ echarts.registerTheme('my_theme', {
-            prevProps.trialsUpdateBroadcast !== this.props.trialsUpdateBroadcast
-        ) {
-            const { columnList } = this.props;
-            this.setState(
-                {
-                    tableColumns: this.initTableColumnList(columnList),
-                    allColumnList: this.getAllColumnKeys()
+    private _buildColumnsFromTableItems(tableItems: any[]): IColumn[] {
+        // extra column, for a icon to expand the trial details panel
+        const columns: IColumn[] = [
+            {
+                key: '_expand',
+                name: '',
+                onRender: (item, index) => {
+                    return (
+                        <Icon
+                            aria-hidden={true}
+                            iconName='ChevronRight'
+                            styles={{
+                                root: {
+                                    transition: 'all 0.2s',
+                                    transform: `rotate(${item._expandDetails ? 90 : 0}deg)`
+                                }
+                            }}
+                            onClick={event => {
+                                event.stopPropagation();
+                                const newItem: any = { ...item, _expandDetails: !item._expandDetails };
+                                const newItems = [...this.state.displayedItems];
+                                newItems[index!] = newItem;
+                                this.setState({
+                                    displayedItems: newItems
+                                });
+                            }}
+                            onMouseDown={e => {
+                                e.stopPropagation();
+                            }}
+                            onMouseUp={e => {
+                                e.stopPropagation();
+                            }}
+                        />
+                    );
                 },
-                () => {
-                    this.updateData();
-                }
-            );
+                fieldName: 'expand',
+                isResizable: false,
+                minWidth: 20,
+                maxWidth: 20
+            }
+        ];
+        // looking at the first row only for now
+        for (const k of Object.keys(tableItems[0])) {
+            const lengths = tableItems.map(item => `${item[k]}`.length);
+            const avgLengths = lengths.reduce((a, b) => a + b) / lengths.length;
+            const columnTitle = _inferColumnTitle(k);
+            const columnWidth = Math.max(columnTitle.length, avgLengths);
+            // TODO: add blacklist
+            columns.push({
+                name: columnTitle,
+                key: k,
+                fieldName: k,
+                minWidth: columnWidth * 4,
+                maxWidth: columnWidth * 10,
+                isResizable: true,
+                onColumnClick: this._onColumnClick.bind(this),
```

Comment:
when page auto refresh, these trials sorted status will be reset. so you could use this function in render(): [example](https://github.com/microsoft/nni/blob/v1.7/src/webui/src/components/trial-detail/TableList.tsx#L582)